instaclustr/resource_cluster.go

```diff
@@ -1655,37 +1681,54 @@ func resourceClusterRead(d *schema.ResourceData, meta interface{}) error {
 	d.Set("pci_compliant_cluster", cluster.PciCompliance == "ENABLED")
 	d.Set("oidc_provider", cluster.OidcProvider)
 
-	azList := make([]string, 0)
-	publicContactPointList := make([]string, 0)
-	privateContactPointList := make([]string, 0)
+	nodesByRack := make(map[string][]Node)
+	publicContactPointsList := make([]string, 0)
+	privateContactPointsList := make([]string, 0)
 
+	// hunt
+	numContactPoints := 0
 	for _, dataCentre := range cluster.DataCentres {
 		for _, node := range dataCentre.Nodes {
-			if !stringInSlice(node.Rack, azList) {
-				if !isDedicatedZookeeperNodeSize(node.Size) {
-					azList = appendIfMissing(azList, node.Rack)
-					privateContactPointList = appendIfMissing(privateContactPointList, node.PrivateAddress)
-					publicContactPointList = appendIfMissing(publicContactPointList, node.PublicAddress)
+			if !isDedicatedZookeeperNodeSize(node.Size) {
+				nodesByRack[node.Rack] = append(nodesByRack[node.Rack], node)
+				numContactPoints++
+			}
+		}
+	}
+
+	// gather
+	sortedRacks := sortedMapKeys(nodesByRack)
+	numRacks := len(sortedRacks)
+	for numContactPoints > 0 {
+		for _, rack := range sortedRacks {
+			nodes := nodesByRack[rack]
+			if len(nodes) > 0 {
```

Comment:
Just out of curiosity could there ever be a rack assigned without a node in it?

Reply:
Unfortunately yes, that code was added after one of the acceptance test created a zookeeper only rack, so we have to handle that case (zookeeper nodesizes are excluded from contact points)